tests/unit/command/test_update.py

```diff
@@ -1,15 +1,36 @@
+import logging
+
 from dvc.cli import parse_args
 from dvc.command.update import CmdUpdate
+from dvc.exceptions import UpdateWithRevNotPossibleError
 
 
 def test_update(dvc_repo, mocker):
     targets = ["target1", "target2", "target3"]
-    cli_args = parse_args(["update"] + targets)
+    cli_args = parse_args(["update", "--rev", "develop"] + targets)
     assert cli_args.func == CmdUpdate
     cmd = cli_args.func(cli_args)
     m = mocker.patch("dvc.repo.Repo.update")
 
     assert cmd.run() == 0
 
-    calls = [mocker.call(target) for target in targets]
+    calls = [mocker.call(target, rev="develop") for target in targets]
     m.assert_has_calls(calls)
+
+
+def test_update_rev_failed(mocker, caplog):
```

Comment:
I don't think this test is valuable. Having `test_update_rev_non_git_failure` above, it only checks that exception outputs it's message. Let's drop it.

Reply:
I agree with @Suor . This test is not harmful, but a more proper one would just raise DvcException and test that CLI wrapper handles that correctly. But we usually neglect those, and it is probably not worth bothering with in this PR. :slightly_smiling_face:

Reply:
done by 2abf41c